### ISSUE TITLE
AIs able to see more than they should

### ISSUE THREAD

**Issue by TribeOfBeavers** [tgstation/tgstation#38857]
Issue reported from  Round ID: 90353 (/tg/Station Bagil [ENGLISH] [US-EAST] [100% FREE LAG])

Round ID: 90353
Testmerges:
#38754: 'Revert "Port AI static to vis_contents"' by optimumtact at commit c19b2bdd60
#38798: 'Refactors component signals to properly register for specific signals' by ninjanomnom at commit 4d864ac9a7
#38436: 'Limb damage rework' by XDTM at commit de9eb262dc
#38831: 'Remove extraneous reagent istype checks, allow carbons (MONKEYS) to get drunk' by vuonojenmustaturska at commit df9d91663c

Reporting client version: 512

An AI reported that they were able to see parts of maint that they shouldn't normally be able to see. I checked the cameras for upgrades with SQL and didn't find anything, nor was I able to see any cameras in areas they aren't normally at. I've included screenshots in case I'm just blind. 

Anyway they specified that they could see: Incinerator, Botany Maintenance, Starboard Quarter Maintenance and the hidden botany. And possibly more but they died before they verify. They said it was probably happening for the entire round, but weren't certain. There were areas with static still, but it was hit and miss.

Screenshots:
https://i.imgur.com/KMOVyRi.png
https://i.imgur.com/jOwxFTE.png

**Comment by SpaceManiac** [comment]
Surprised that this happened with the #38754 testmerge active, I'm still trying to investigate the real issue (brought up in #38734) but that should've prevented it in the meantime.

**Comment by fizzle7** [comment]
Ignore that, I got random static all over xeno, this is shit.

I've been testing the xenobio console for static and it doesn't show static in maint when you go to the wall with xeno maint. However, if you move a certain distance away, static starts showing in part of maint but not in the rest, the results are the following:

It doesn't necessarily all behave like complete static, instead it's spotty similar to the random static bug, but in this case the static should actually be there. If you alt-tab sometimes more static spots appear where they should be, but they disappear the next time you open the console (all static disappears when you open the console so you have to force it to reappear by the method detailed below, however, only alt-tabbing bring out new static that couldn't be brought with the method).Now whether it's in spots or complete static, it does obey a rule. Whenever you move the camera's centre 17 tiles away horizontally or vertically, counting the border static tile as 0 (that is, the last static tile after which there is no static), the static appears horizontally or vertically, mostly spotty static, not complete. So, if for example maint is west of xeno, and you move 17 tiles away east, static will appear in the western part of maint, but it will always be 17 tiles away.

**Comment by Irafas** [comment]
When I was playing AI I was able to see into securities execution room

![f533e4b40b0e960d2699a84b1e7b8e9a](https://user-images.githubusercontent.com/26196397/42233357-3cd8eea4-7ee9-11e8-884c-b6ac2c92d142.png)

I believe it was the same testmerges

**Comment by fizzle7** [comment]
Yep I can see that too, also I just saw static as ai that is supposed to be there suddenly appear while in view and then a few minutes later gone when I checked back

**Comment by BebeYoshi** [comment]
An AI main claimed to be able to see the entire lavaland, as long as there was lighting, of course, they also stated this way to reproduce the issue "Tends to work when I jump to labor network "

**Comment by actioninja** [comment]
Still an issue, still coming up. There are a couple issue reports at this time about this, closing them and linking them here.

**Comment by redmoogle** [comment]
Can confirm the bug is still happening on Sybil

**Comment by capn-monkeypaw** [comment]
Multiple reports about this continuing to be an issue on Events [EU].

**Comment by redguy999** [comment]
switching to multi-camera view and back has sometimes caused this, changing to another network has also caused this.

I believe this has something to do with how static tiles work, either the tiles themselves, or why they appear to the AI in the first place.

**Comment by mudzbe** [comment]
This happens a lot, I could see everywhere, space too, although, I stayed on Z level when reaching a map border. It fixed itself when I jumped to someone through the camera tracking menu. I don't know what causes it.

**Comment by Kylerace** [comment]
the only way this should be able to happen is if somehow the global cameranet static images arent being added to the ai's client. each of those images are embedded inside of the cameranet invisible objects added to the vis_contents of all tiles an ai moves over and not in view of a cameranet. the vis_contents are never removed unless the camerachunk itself updates and all the tiles are confirmed visible, so the only thing that makes sense is that the ai player isnt getting those two global images added to their client.images
